app/code/community/Razorpay/Payments/Model/Paymentmethod.php

```diff
@@ -110,55 +110,17 @@ public function captureOrder(Varien_Object $order, $amount)
         return [$result, $error];
     }
 
-    public function capturePayment($response)
+    public function validateSignature($response)
     {
-        $order = Mage::getModel('sales/order');
-        $payment = Mage::getModel('sales/order_payment');
-        $order->loadByIncrementId($response['magento_order_id']);
-
-        $payment->setOrder($order);
-
+        // I have to use the order_id from the server and not the response object.
+        
+        $signature = hash_hmac('sha256', Mage::getSingleton('core/session')->getRazorpayOrderID() . "|" . $response['razorpay_payment_id'], $this->getConfigData('key_secret'));
+        
         $success = false;
 
-        $paymentId = null;
-
-        if (empty($response['razorpay_payment_id']) === false)
-        {
-            $paymentId = $response['razorpay_payment_id'];
-
-            try
-            {
-                $amount = $order->getBaseGrandTotal();
-
-                list($success, $error) = $this->captureOrder($payment, $amount);
-            }
-            catch (\Exception $e)
-            {
-                $success = false;
-            }
-        }
-
-        if ($success === true)
+        if (hash_equals($signature , $response['razorpay_signature']))
         {
-            $order->sendNewOrderEmail();
```

Comment:
@mayankamencherla please add this part back. We need to store the Razorpay payment id in the order logs for reference. Everything else is fine.